aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -273,13 +278,24 @@ private boolean isGlueIcebergTable(Table table) {
   public boolean dropTable(TableIdentifier identifier, boolean purge) {
     try {
       TableOperations ops = newTableOps(identifier);
-      TableMetadata lastMetadata = ops.current();
+
+      GlueTableOperations glueOps = (GlueTableOperations) ops;
+      S3FileIO s3FileIO = (S3FileIO) glueOps.io();
+      TableMetadata lastMetadata = null;
+      boolean isTablePurged = isTablePurged(identifier, s3FileIO.client());
+      if (!isTablePurged) {
+        lastMetadata = ops.current();
+      }
+
       glue.deleteTable(DeleteTableRequest.builder()
           .catalogId(awsProperties.glueCatalogId())
           .databaseName(IcebergToGlueConverter.getDatabaseName(identifier))
           .name(identifier.name())
           .build());
       LOG.info("Successfully dropped table {} from Glue", identifier);
+      ValidationException.check(!purge || !awsProperties.glueLakeFormationEnabled() || isTablePurged,
+          "Cannot purge table with LakeFormation enabled because S3 access is lost after table is dropped. " +
```

Comment:
Overall for this section of logic, my opinion is that we should not do glue.deleteTable if the table is LF enabled and purge is requested. So the validation should happen before we do the delete.

Reply:
I think we still want the glue table to be dropped even LF enabled and purge is set to true? And the original logic is to ensure the glue table is dropped even purge fails.

Reply:
> I think we still want the glue table to be dropped even LF enabled and purge is set to true?

Let's think about this again...

What we know is that if table is LF enabled and purge is true, there is no way we can make this happen safely, because:
1. if we drop table first, we can pre-fetch s3 credential but that might not be enough time to remove all the table data files
2. if we remove all the table data files first, we might fail in the middle, this causes orphan files. 

Originally I suggested that because there is no perfect way, let's just ask customer to purge by themselves and we validate and not allow them to purge in Iceberg if LF is enabled. But now I think again, this is known behavior even for other normal tables, and it looks like we should go with 2, so that we first remove table data files, all the failures are already suppressed in the operation, and then we drop the table.

+ @amogh-jahagirdar any thoughts?

Reply:
Sorry for the late reply, but yeah Pre-fetching credentials and deleting the table is too brittle because those credentials would be invalidated. However, the case I am thinking of is if we do a best effort deletion of the files first, if there's concurrent operations which don't conflict (like append), those could succeed. Then if dropping the table after fails, then the table is in a bad state. To address this there can be a DROP_IN_PROGRESS STATE introduced which is set first and then used when committing to prevent commits against a table being deleted. I think any attempt of deletion of files should only happen when it's impossible to modify the table further.

Let me know if I misunderstood the problem though, I'll think more on it but I think 1 certainly can be ruled out.

Reply:
Discussed offline, we decide to go with approach 1, a few reasons:
1. we don't want the data file to be deleted if the users don't have the drop permission, so this rules out the option 2
2. we can use the pre-fetched credentials because LF always gives us the credentials for each individual service, and it is always our own responsibility to use them in a disciplined way. Even after the Glue table is dropped, we know those S3 files belong to this table, so using the pre-fetched S3 credentials to delete them is OK. The credential can expire, but the file deletion is best effort anyway.